easybuild/toolchains/compiler/craypewrappers.py

```diff
@@ -0,0 +1,182 @@
+
+
+class CrayPEWrapper(Compiler):
+    """Generic support for using Cray compiler wrappers"""
+
+    # no toolchain components, so no modules to list here (empty toolchain definition w.r.t. components)
+    # the PrgEnv and craype are loaded, but are not considered actual toolchain components
+    COMPILER_MODULE_NAME = []
+    COMPILER_FAMILY = TC_CONSTANT_CRAYPEWRAPPER
+
+    COMPILER_UNIQUE_OPTS = {
+        # FIXME: (kehoste) how is this different from the existing 'shared' toolchain option? just map 'shared' to '-dynamic'? (already done)
+        'dynamic': (True, "Generate dynamically linked executables and libraries."),
+        'mpich-mt': (False, "Directs the driver to link in an alternate version of the Cray-MPICH library which \
+                             provides fine-grained multi-threading support to applications that perform \
+                             MPI operations within threaded regions."),
+        'usewrappedcompiler': (False, "Use the embedded compiler instead of the wrapper"),
+    }
+
+    COMPILER_UNIQUE_OPTION_MAP = {
+        'pic': 'shared',
+        'shared': 'dynamic',
+        'static': 'static',
+        'verbose': 'craype-verbose',
+        'mpich-mt': 'craympich-mt',
+    }
+
+    COMPILER_CC = 'cc'
+    COMPILER_CXX = 'CC'
+
+    COMPILER_F77 = 'ftn'
+    COMPILER_F90 = 'ftn'
+
+    # FIXME (kehoste) hmmmm, really? then how do you control optimisation, precision when using the Cray wrappers?
+    COMPILER_FLAGS = []  # we dont have this for the wrappers
+    COMPILER_OPT_FLAGS = []  # or those
+    COMPILER_PREC_FLAGS = []  # and those for sure not !
+
+    # template and name suffix for PrgEnv module that matches this toolchain
+    # e.g. 'gnu' => 'PrgEnv-gnu/<version>'
+    PRGENV_MODULE_NAME_TEMPLATE = 'PrgEnv-%(suffix)s/%(version)s'
+    PRGENV_MODULE_NAME_SUFFIX = None
+
+    # template for craype module (determines code generator backend of Cray compiler wrappers)
+    CRAYPE_MODULE_NAME_TEMPLATE = 'craype-%(optarch)s'
+
+    def _pre_preprare(self):
+        """Load PrgEnv module."""
+        prgenv_mod_name = self.PRGENV_MODULE_NAME_TEMPLATE % {
+            'suffix': self.PRGENV_MODULE_NAME_SUFFIX,
+            'version': self.version,
+        }
+        self.log.info("Loading PrgEnv module '%s' for Cray toolchain %s" % (prgenv_mod_name, self.mod_short_name))
+        self.modules_tool.load([prgenv_mod_name])
+
+    def _set_optimal_architecture(self):
+        """Load craype module specified via 'optarch' build option."""
+        optarch = build_option('optarch')
+        if optarch is None:
+            self.log.error("Don't know which 'craype' module to load, 'optarch' build option is unspecified.")
+        else:
+            self.modules_tool.load([self.CRAYPE_MODULE_NAME_TEMPLATE % {'optarch': optarch}])
+
+    # FIXME: (kehoste) is it really needed to customise this?
+    # this looks like a workaround for setting the COMPILER_*_FLAGS lists empty?
+    def _set_compiler_flags(self):
```

Comment:
it kinda is, because the current Compiler base class is too specific for GCC like compilers.